src/modules/70-pipeline/components/PipelineStudio/SavePipelinePopover/SavePipelinePopover.tsx

```diff
@@ -0,0 +1,452 @@
+          showErrorModal()
+        } else {
+          showError(
+            getRBACErrorMessage({ data: response as AccessControlCheckError }) || getString('errorWhileSaving'),
+            undefined,
+            'pipeline.save.pipeline.error'
+          )
+        }
+      } else {
+        // isGitSyncEnabled true
+        throw response
+      }
+    }
+    return { status: response?.status }
+  }
+
+  const saveAngPublishWithGitInfo = async (
+    updatedGitDetails: SaveToGitFormInterface,
+    payload?: SavePipelineObj,
+    objectId?: string
+  ): Promise<UseSaveSuccessResponse> => {
+    let latestPipeline: PipelineInfoConfig = payload?.pipeline || pipeline
+
+    if (isYaml && yamlHandler) {
+      try {
+        latestPipeline = payload?.pipeline || (parse(yamlHandler.getLatestYaml()).pipeline as PipelineInfoConfig)
+      } /* istanbul ignore next */ catch (err) {
+        showError(err.message || err, undefined, 'pipeline.save.gitinfo.error')
+      }
+    }
+
+    const response = await saveAndPublishPipeline(
+      latestPipeline,
+      omit(updatedGitDetails, 'name', 'identifier'),
+      pipelineIdentifier !== DefaultNewPipelineId ? { lastObjectId: objectId } : {}
+    )
+
+    return {
+      status: response?.status
+    }
+  }
+
+  const { openSaveToGitDialog } = useSaveToGitDialog<SavePipelineObj>({
+    onSuccess: (
+      gitData: SaveToGitFormInterface,
+      payload?: SavePipelineObj,
+      objectId?: string
+    ): Promise<UseSaveSuccessResponse> =>
+      saveAngPublishWithGitInfo(gitData, payload, objectId || gitDetails?.objectId || '')
+  })
+
+  const saveAndPublish = React.useCallback(async () => {
+    window.dispatchEvent(new CustomEvent('SAVE_PIPELINE_CLICKED'))
+
+    let latestPipeline: PipelineInfoConfig = pipeline
+
+    if (isYaml && yamlHandler) {
+      if (!parse(yamlHandler.getLatestYaml())) {
+        clear()
+        showError(getString('invalidYamlText'))
+        return
+      }
+      try {
+        latestPipeline = parse(yamlHandler.getLatestYaml()).pipeline as PipelineInfoConfig
+      } /* istanbul ignore next */ catch (err) {
+        showError(err.message || err, undefined, 'pipeline.save.pipeline.error')
+      }
+    }
+
+    const ciCodeBaseConfigurationError = validateCICodebaseConfiguration({ pipeline: latestPipeline, getString })
+    if (ciCodeBaseConfigurationError) {
+      clear()
+      showError(ciCodeBaseConfigurationError)
+      return
+    }
+
+    // if Git sync enabled then display modal
+    if (isGitSyncEnabled) {
+      if (isEmpty(gitDetails.repoIdentifier) || isEmpty(gitDetails.branch)) {
+        clear()
+        showError(getString('pipeline.gitExperience.selectRepoBranch'))
+        return
+      }
+      // When git sync enabled, do not irritate user by taking all git info then at the end showing BE errors related to schema
+      const error = await validateJSONWithSchema({ pipeline: latestPipeline }, pipelineSchema?.data as any)
```

Comment:
Codacy has a fix for the issue: [Unexpected any. Specify a different type.](https://app.codacy.com/gh/harness/harness-core-ui/pullRequest?prid=9232314)
```suggestion
      const error = await validateJSONWithSchema({ pipeline: latestPipeline }, pipelineSchema?.data as unknown)
```